easybuild/easyconfigs/p/PyTorch/PyTorch-1.10.1-CrayGNU-21.09.eb

```diff
@@ -0,0 +1,85 @@
+# contributed by Luca Marsella (CSCS)
+easyblock = 'PythonBundle'
+
+name = 'PyTorch'
+version = '1.10.1'
+_ptcudaver = 'cu113'
+
+homepage = 'https://pytorch.org'
+description = """Tensors and Dynamic neural networks in Python with strong GPU acceleration.
+PyTorch is a deep learning framework that puts Python first."""
+
+toolchain = {'name': 'CrayGNU', 'version': '21.09'}
+toolchainopts = {'pic': True, 'verbose': False}
+
+builddependencies = [
+    ('cudatoolkit/11.2.0_3.39-2.1__gf93aa1c', EXTERNAL_MODULE),
+    ('libffi', '3.4.2'),
+    ('wheel', '0.37.0')
+]
+
+dependencies = [
+    ('Boost', '1.77.0', '-python%(pymajver)s'),
+    ('cray-python', EXTERNAL_MODULE)
+]
+
+preinstallopts = ' export HOROVOD_BUILD_CUDA_CC_LIST=60; '
+preinstallopts += ' export HOROVOD_CUDA_HOME=/usr/local/cuda-11.2; '
+preinstallopts += ' export HOROVOD_GPU_OPERATIONS=NCCL; '
+preinstallopts += ' export HOROVOD_MPICXX_SHOW="CC --cray-print-opts=all"; '
+preinstallopts += ' export HOROVOD_NCCL_HOME=/opt/nvidia/hpc_sdk/Linux_x86_64/21.3/comm_libs/nccl; '
+preinstallopts += ' export HOROVOD_NCCL_LINK=SHARED; '
+preinstallopts += ' export HOROVOD_WITHOUT_MXNET=1; '
+preinstallopts += ' export HOROVOD_WITH_PYTORCH=1; '
+preinstallopts += ' export HOROVOD_WITHOUT_TENSORFLOW=1; '
+
+sanity_pip_check = True
+exts_defaultclass = 'PythonPackage'
+exts_default_options = {
+    'req_py_majver': '%(pymajver)s',
+    'req_py_minver': '%(pyminver)s',
+    'source_urls': [PYPI_SOURCE],
+    'unpack_sources': False,
+    'use_pip': True
+}
+
+exts_list = [
+    ('cffi', '1.15.0'),
+    ('cloudpickle', '2.0.0'),
+    ('iniconfig', '1.1.1'),
+    ('Pillow', '8.4.0', {'modulename': 'PIL'}),
+    ('psutil', '5.8.0'),
+    ('pycparser', '2.21'),
+    ('PyYAML', '6.0', {'modulename': 'yaml'}),
+    ('pytest', '6.2.5', {
+        'source_tmpl': '%(name)s-%(version)s-py%(pymajver)s-none-any.whl'
+    }),
+    ('toml', '0.10.2'),
```

Comment:
TOML? :) out of curiosity, but where does this comes from?
pytest is probably also not necessary... but scipy is for the reframe tests
I'm trying a similar recipe on 20.11 and can share it later on

Reply:
I had to add `toml` otherwise the build was failing, same for `pytest`. The package `scipy` is provided by `cray-python` instead and the ReFrame checks triggered by `pytorch_horovod_check.py` passed with my local modulefile on Dom.

Reply:
Fair enough, thanks!
if you have the module built, please make sure the reframe tests run with the provided version of scipy (I guess it should tho)